packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -358,5 +358,10 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection impleme
             // If it's not critical error, we want to raise event on this object only.
             this.emit("disconnect", reason);
         }
+
+        // We set the closed flag as a part of the contract for overriding the disconnect method.  This is used by
+        // DocumentDeltaConnection to determine if emitting on the socket is allowed, which is important since
+        // OdspDocumentDeltaConnection reuses the socket rather than truly disconnecting it.
+        this.closed = true;
```

Comment:
i don't love this. i'm alway wary of flag as they make the state machine harder to understand. I also don't like having to  reach into the shared implementation. Howerver, if we think this will get better soon-ish with refactoring i think it's ok.

Reply:
My hope is this all gets simpler with refactoring or `BatchManager` removal per #3788.  Agreed on wanting to minimize state management.